### PR TITLE
Stack and queue

### PR DESCRIPTION
**Pull request by catdude2000**
- [x] Top-level README “Table of Contents” is updated
 - [x] README for this challenge is complete
       - [ ] Summary, Description, Approach & Efficiency, Solution
       - [ ] Picture of whiteboard
       - [ ] Link to code
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
       - [ ] “Happy Path” - Expected outcome
       - [ ] Expected failure
       - [ ] Edge Case (if applicable/obvious)

### CODE REVIEW

File: javascript/linked-list/stacks-and-queues/stacksandqueues.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    this.top = temp.next;
+    temp.next = null;
+    return temp; //or just temp
+  }
+  peek() {
+    return this.top.value;
+  }
+  isEmpty() {
+    return this.top = null;
+  }
```
Comment (DSFrey) [inline javascript/linked-list/stacks-and-queues/stacksandqueues.js:37]: One `=` is the assignment operator. This needs to be an equality operator.

File: javascript/linked-list/stacks-and-queues/stacksandqueues.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  peek() {
+    if (!this.front) {
+      return null;
+    } else {
+      return this.front.value;
+    }
+  }
+  isEmpty() {
+    return this.front = null;
+  }
```
Comment (DSFrey) [inline javascript/linked-list/stacks-and-queues/stacksandqueues.js:72]: Again, wrong operator.

File: javascript/linked-list/stacks-and-queues/stacksandqueues.test.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    expect(testStack.top.value).toEqual('dummyNode');
+  });
+  test('Can successfully empty a stack after multiple pops', () => {
+    const testStack = new Stack(new Node('dummyNode'));
+    testStack.push('dummyNodeStacker');
+    testStack.push('dummyNodeStacker2');
+    testStack.pop('dummyNodeStacker2');
+    testStack.pop('dummyNodeStacker');
+    expect(testStack.isEmpty).toBeTruthy();
+  });
```
Comment (DSFrey) [inline javascript/linked-list/stacks-and-queues/stacksandqueues.test.js:31]: Your `pop()` method does not need an argument.

Also, in the empty a stack test you are creating your stack with a node then pushing two more, but you are only popping twice.

File: javascript/linked-list/stacks-and-queues/stacksandqueues.test.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    testStack.pop('dummyNodeStacker');
+    expect(testStack.top.value).toEqual('dummyNode');
+  });
+  test('Can successfully empty a stack after multiple pops', () => {
+    const testStack = new Stack(new Node('dummyNode'));
+    testStack.push('dummyNodeStacker');
+    testStack.push('dummyNodeStacker2');
+    testStack.pop('dummyNodeStacker2');
+    testStack.pop('dummyNodeStacker');
+    expect(testStack.isEmpty).toBeTruthy();
```
Comment (DSFrey) [inline javascript/linked-list/stacks-and-queues/stacksandqueues.test.js:30]: You are not calling the method here because you are missing the parentheses.
```js
expect(testStack.isEmpty()).toBeTruthy();

File: javascript/linked-list/stacks-and-queues/stacksandqueues.test.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    testQueue.enqueue('dummyNodeQueuer2');
+    expect(testQueue.rear.value).toEqual('dummyNodeQueuer2');
+  });
+  test('Can successfully dequeue out of a queue the expected value', () => {
+    const testQueue = new Queue();
+    testQueue.enqueue('dummyNodeQueuer');
+    testQueue.enqueue('dummyNodeQueuer2');
+    testQueue.dequeue('dummyNodeQueuer');
+    // console.log(testQueue, 'testqueue');
+    expect(testQueue.rear.value).toEqual('dummyNodeQueuer2');
```
Comment (DSFrey) [inline javascript/linked-list/stacks-and-queues/stacksandqueues.test.js:63]: Again, `dequeue()`should not have an argument.

File: javascript/linked-list/stacks-and-queues/stacksandqueues.test.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  });
+  test('Can successfully instantiate an empty queue', () => {
+    const testQueue = new Queue();
+    expect(testQueue.isEmpty()).toEqual(null);
+  });
+  test('Calling dequeue or peek on empty queue raises exception', () => {
+    const testQueue = new Queue();
+    // expect(testQueue.peek()).toThrow(Error);
+    expect(testQueue.peek()).toEqual(null);
+  });
```
Comment (DSFrey) [inline javascript/linked-list/stacks-and-queues/stacksandqueues.test.js:85]: You are not throwing an error because your method is telling it to return null. If you want to throw it will look like this:
```js
  peek() {
    if (!this.front) {
      throw new Error('Queue is empty');
    } else {
      return this.front.value;
    }
  }
```
Then your test needs to take in a callback so it doesn't error out when you invoke:
```js
expect(testQueue.peek).toThrow('Queue is empty');
// or
expect(() => testQueue.peek()).toThrow('Queue is empty')
```
